include/swaybar/config.h

```diff
@@ -65,6 +76,16 @@ uint32_t parse_position(const char *position);
 char *parse_font(const char *font);
 
 /**
+ * Parse display mode dock|hide|invisible.
+ */
+enum display_mode_types parse_display_mode(const char *display_mode);
+
+/**
+ * Parse hidden state show|hide.
+ */
+enum hidden_states parse_hidden_state(const char *hidden_state);
+
```

Comment:
> Take it on faith that I know what I'm doing.

Hell no. If you can't/won't defend a contrary viewpoint, I'm not going to take that as proof you're right. But I do accept that in your project you get to decide such things (as I do in mine).

> Well, if the variable is an anonymous struct,

By which you mean "But I implemented enum->string using anonymous structs in 60ce81e".
Well, my view is that you probably shouldn't have. Translating an enum into a string is a common operation so relying on an anon struct is a doubtful choice.